Урок 2. Дискретные распределения вероятностей
Условие:

Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.
Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?
Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?
В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?


 1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.
            Формула биномиального распределения: P(n)(X=k) = С(n,k) * p^k * q^(n-k)
            k = 85
            n = 100
            p = 0.8
            q = 1-0.8 = 0.2

In [3]:
from math import factorial
a = factorial(100) // (factorial (85) * factorial (100-85) )
b = 0.8**85
v = 0.2**(100-85)

a, b, v

(253338471349988640, 5.789604461865837e-09, 3.276800000000003e-11)

 cсоответственно подставляем данные :
            a -  это С(n,k)
            b - это p^k 
            v - это q^(n-k)
             P(n)(X=k) = С(n,k) * p^k * q^(n-k)


In [4]:
a*b*v


0.048061793700746556

вероятность того, что стрелок попадет в цель ровно 85 раз   = 0.048061793700746556

2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

             1. Какова вероятность, что ни одна из них не перегорит в первый день?
            будем вычислять через формудлу Пуассона  
            P(m) = (lambda^m / m!) * e^(-lambda)
            для начала определим lambda

In [7]:
p = 0.0004
n = 5000
lamda = p * n 
lamda

2.0

In [8]:
import math
P_0 = (lamda**0/(factorial(0))) * (math.e **-lamda)
P_0

0.1353352832366127

вероятность того, что ни одна из них не перегорит в первый день = 0.1353352832366127 или 13,53%

            2. Какова вероятность, что перегорят ровно две?

In [9]:
P_2 = (lamda**2/(factorial(2))) * (math.e **-lamda)
P_2

0.2706705664732254

вероятность того, что перегорят ровно две  = 0.2706705664732254 или 27,07%


3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

In [51]:
n = 144
k = 70
p = 0.5 # вероятность при выпадении орла 70 раз 
p

0.5

In [46]:
p = 0.5
q= 1 - p  #определяем обратную вероятность
q

0.5

Далее ищем сочетание C(n,k)

In [52]:
a = factorial(144) // (factorial (70) * factorial (144-70) )
a

1400749509083708812152608657850239525991600

Найдем p^k и q^(n-k)

In [53]:
b = p ** k
v = q ** (n-k)
b,v

(8.470329472543003e-22, 5.293955920339377e-23)

посчитали данные  и подставляем в формулу : P(n)(X=k) = C(n,k) * p^k * q^(n-k)

In [54]:
a * b * v

0.06281178035144776

вероятность, что орел выпадет ровно 70 раз равна 0.06281178035144776или 6.28 % 

В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?
    1. Какова вероятность того, что все мячи белые?
    расчитываем вероятность вытащить 2 мяча в первом ящике.

In [3]:
from math import factorial
a = factorial(7) // (factorial (2) * factorial (7-2) )
b = factorial(3) // (factorial (0) * factorial (3-0) )
c = factorial(10) // (factorial (2) * factorial (10-2) )
a,b,c

(21, 1, 45)

In [5]:
P_1 = (a*b)/c
P_1

0.4666666666666667

            вероятность вытащить 2 мяча в первом ящике = 0.4666

             расчитываем вероятность вытащить 2 мяча во втором ящике.

In [9]:

d = factorial(9) // (factorial (2) * factorial (9-2) )
f = factorial(2) // (factorial (0) * factorial (2-0) )
g = factorial(11) // (factorial (2) * factorial (11-2) )
d,f,g

(36, 1, 55)

In [11]:
P_2 = (d*f)/g
P_2

0.6545454545454545

            вероятность вытащить 2 мяча во втором ящике =  0.6545

    определяем вероятность что все мячи белые

In [14]:
P_1 = 0.4666666666666667
P_2 = 0.6545454545454545
P = P_1 * P_2
P

0.3054545454545455

вероятность что все мячи белые = 0.3054

    2. Какова вероятность того, что ровно два мяча белые?

In [34]:
w = factorial(7) // (factorial (2) * factorial (7-2) )
e = factorial(3) // (factorial (0) * factorial (3-0) )

R = factorial(10) // (factorial (2) * factorial (10-2)) 

t = factorial(7) // (factorial (0) * factorial (7-0) )
y = factorial(3) // (factorial (2) * factorial (3-2) )

Y  = R

u = factorial(7) // (factorial (1) * factorial (7-1) )
i = factorial(3) // (factorial (1) * factorial (3-1) )
O = R
# R для трех исходов одинаковый

c_1 =(e*w)/R
c_2 =(t*y)/R
c_3 =(u*i)/R



p = factorial(2) // (factorial (2) * factorial (2-2) )
s = factorial(9) // (factorial (0) * factorial (9-0) )

H = factorial(11) // (factorial (2) * factorial (11-2)) 

j = factorial(9) // (factorial (2) * factorial (9-2) )
n = factorial(2) // (factorial (0) * factorial (2-0) )

k = factorial(9) // (factorial (1) * factorial (9-1) )
l = factorial(2) // (factorial (1) * factorial (2-1) )

# H для трех исходов одинаковый
c_4 =(p*s)/H
c_5 =(j*n)/H
c_6 =(k*l)/H

c_1, c_2, c_3, c_4, c_5, c_6 

(0.4666666666666667,
 0.06666666666666667,
 0.4666666666666667,
 0.01818181818181818,
 0.6545454545454545,
 0.32727272727272727)

In [35]:
P_all = (c_1*c_4) + (c_2 *c_5) + (c_3 * c_6)
P_all


0.20484848484848486

вероятность того, что ровно два мяча белые  = 0.2048

    3. Какова вероятность того, что хотя бы один мяч белый?
    будем исходить от обратного чтобы избежать кучи расчетов(пример из лекции)
    найдем исход выпадения всех черных 
    

In [38]:
T = factorial(7) // (factorial (0) * factorial (7-0) )
Y = factorial(3) // (factorial (2) * factorial (3-2) )

P_TY = (T*Y)/R
P_TY 

0.06666666666666667

In [41]:
S = factorial(9) // (factorial (0) * factorial (9-0) )
P = factorial(2) // (factorial (2) * factorial (2-2) )

P_SP = (S*P)/H
P_SP

0.9818181818181818

In [43]:
P_TYSP =P_TY * P_SP
P_TYSP

0.0012121212121212121

In [45]:
Q = 1-P_TYSP
Q

0.9987878787878788

вероятность того, что хотя бы один мяч белый посчитана   = 0.9987